dynamic_rest/serializers.py

```diff
@@ -346,6 +361,22 @@ def get_fields(self):
         for name in deferred:
             serializer_fields.pop(name)
 
+        # Set read_only flags based on read_only_fields meta list.
+        # Here to cover DynamicFields not covered by DRF.
+        ro_fields = getattr(self.Meta, 'read_only_fields', [])
+        self.make_fields_read_only(serializer_fields, ro_fields, True)
+
+        # Toggle read_only flags for immutable fields.
+        immutable_field_names = self._get_flagged_field_names(
+            serializer_fields,
+            'immutable'
+        )
+        self.make_fields_read_only(
```

Comment:
Actually this was intentional. One case we want this today is with UUID `id` fields, which I believe DRF automatically infers as a read-only field, but in some cases want to allow the client to generate IDs and set on POST. This approach allows us to override the field and make it immutable, without having to explicitly declare the field.